src/main.ts

```diff
@@ -14,30 +14,26 @@ export default class EditInNeovim extends Plugin {
   async onload() {
     await this.loadSettings();
     this.pluginChecks();
-
-    this.neovim = new Neovim(this.settings);
-    const adapter = this.app.vault.adapter as FileSystemAdapter;
-
-    if (this.settings.openNeovimOnLoad) this.neovim.newInstance(adapter);
+    await this.initNeovim();
 
     this.registerEvent(
       this.app.workspace.on("file-open", this.neovim.openFile),
     );
-
     this.registerEvent(
       this.app.workspace.on("quit", this.neovim?.close),
     );
 
-    // This adds a settings tab so the user can configure various aspects of the plugin
```

Comment:
While not necessarily needed, I'm not sure why you removed this, can you explain it for me please?